### PR TITLE
Generate CLIs with Cobra

### LINKED ISSUE #30: Switch command line utilities to use cobra
This is a tracking ticket for upgrading this repo to use [cobra](https://github.com/spf13/cobra). I think we probably want to produce a single `strest-grpc` CLI, with `client` and `server` subcommands for running the client and the server. But also totally open to suggestion on that.

### PR DESCRIPTION
**Pull request by hawkw**
CLIs are now implemented using the `cobra` library. `strest-grpc` now has `client` and `server` subcommands, the flags for which are the same as before.

Note that the way this is implemented is pretty ugly - I turned the `main()` functions in `client` and `server` into `Run()` and just had cobra pass all the flags in as arguments. Probably this could be made more idiomatic, either using `viper` in addition to `cobra` or by passing arguments in a struct, but this Basically Works.

Follow-ons:
- [ ] make it less ugly
- [ ] CLI design/UX pass
- [ ] viper?

Closes #30

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+var RootCmd = &cobra.Command{
+	Use:   "strest-grpc [client | server]",
```
Comment (siggy) [inline cmd/root.go]: `max-rps` ?

File: cmd/client.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	"github.com/buoyantio/strest-grpc/client"
+	"github.com/spf13/cobra"
+)
+
+var clientCfg = client.Config{}
+
+var clientCmd = &cobra.Command{
+	Use:   "client",
+	Short: "run the strest-grpc client",
+	Args:  cobra.ArbitraryArgs,
```
Comment (siggy) [inline cmd/client.go]: noticed the other commands use `cobra.NoArgs`? is this related to how we're doing `clientCfg.Run()` but also `server.Run()`. curious about the motivation for different patterns between client and server commands.

Comment (hawkw) [reply cmd/client.go]: Ah, this is a mistake on my part - it's supposed to be `NoArgs`

**Comment by hawkw** [comment]
@siggy I think I've made the changes you requested; we may, however, eventually want to change to letting Cobra generate our documentation for us to ensure it stays up to date.

**Pull request merged**